packages/base-controller/src/RestrictedControllerMessenger.ts

```diff
@@ -95,8 +111,7 @@ export class RestrictedControllerMessenger<
   registerActionHandler<
     ActionType extends Action['type'] & NamespacedName<Namespace>,
   >(action: ActionType, handler: ActionHandler<Action, ActionType>) {
-    /* istanbul ignore if */ // Branch unreachable with valid types
```

Comment:
Looks like this is responsible for the drop in coverage. What was the reasoning behind removing this?

As a general practice, ignoring unreachable code (or code that can't be reached with valid types) seems.... generally OK to me. It's certainly less important than other code, because it's outside the scope of what the API is intended to do.

But it does seem like a code smell, because if a branch is unreachable, why does it exist? If code is here, it's probably because it does something we care about.

This case is in a grey area like that; this was written this way because I knew this was used in a JavaScript context. So in this case it's probably better to test it, so this decision to remove the ignore comment seems appropriate.

But still, I'm curious to hear your thoughts.

Reply:
It looks like our clients are calling these methods in JavaScript environments, so I think we definitely want both compile time and runtime assurances on these branches.

I added tests here https://github.com/MetaMask/core/pull/2051/commits/93e1168528643d3fa9243ae9ca35688cfc29bae5 that bring the jest coverage to 100% on all fronts.

Reply:
I extracted the tests to a separate PR: https://github.com/MetaMask/core/pull/2058